pkg/guacrest/helpers/getPackageInfo.go

```diff
@@ -0,0 +1,250 @@
+
+		var dependencies []gen.PackageInfo
+
+		deps, err := searchDependencies(ctx, gqlClient, pkgSpec, searchSoftware, *latestSbom)
+		if err != nil {
+			return nil, err
+		}
+		for _, purl := range deps {
+			dependencies = append(dependencies, purl)
+		}
+		response.Dependencies = &dependencies
+	}
+
+	return &response, nil
+}
+
+// searchAttachedVulns searches for vulnerabilities associated with the given package
+// and its dependencies via a BFS.
+//
+// Parameters:
+// - ctx: The context for the operation
+// - gqlClient: The GraphQL client used for querying
+// - pkgSpec: The package specification to start the search from
+//
+// Returns:
+// - A slice of Vulnerability objects containing the found vulnerabilities
+// - An error
+//
+// The function performs a breadth-first search starting from the given package,
+// collecting vulnerabilities for each package and its dependencies.
+func searchAttachedVulns(ctx context.Context, gqlClient graphql.Client, pkgSpec model.PkgSpec, searchSoftware bool, startSBOM model.AllHasSBOMTree) ([]gen.Vulnerability, error) {
+	logger := logging.FromContext(ctx)
+	var vulnerabilities []gen.Vulnerability
+
+	pkgs, err := searchDependencies(ctx, gqlClient, pkgSpec, searchSoftware, startSBOM)
+	if err != nil {
+		return nil, fmt.Errorf("error searching dependencies: %w", err)
+	}
+
+	for pkg, purl := range pkgs {
+		logger.Infof("package: %+v, purl: %s", pkg, purl)
+	}
+
+	for pkg := range pkgs {
+		vulns, err := model.CertifyVuln(ctx, gqlClient, model.CertifyVulnSpec{
+			Package: &model.PkgSpec{
+				Id: &pkg,
+			},
+		})
+		if err != nil {
+			return nil, fmt.Errorf("error fetching vulnerabilities from package spec: %w", err)
+		}
+
+		for _, vuln := range vulns.CertifyVuln {
+			vulnerability := gen.Vulnerability{
+				Metadata: gen.ScanMetadata{
+					Collector:      &vuln.Metadata.Collector,
+					DbUri:          &vuln.Metadata.DbUri,
+					DbVersion:      &vuln.Metadata.DbVersion,
+					Origin:         &vuln.Metadata.Origin,
+					ScannerUri:     &vuln.Metadata.ScannerUri,
+					ScannerVersion: &vuln.Metadata.ScannerVersion,
+					TimeScanned:    &vuln.Metadata.TimeScanned,
+				},
+				Vulnerability: gen.VulnerabilityDetails{
+					Type: &vuln.Vulnerability.Type,
+				},
+			}
+
+			for _, namespace := range vuln.Package.Namespaces {
+				for _, name := range namespace.Names {
+					for _, version := range name.Versions {
+						vulnerability.Packages = append(vulnerability.Packages, version.Purl)
+					}
+				}
+			}
+
+			for _, vIDs := range vuln.Vulnerability.VulnerabilityIDs {
+				vulnerability.Vulnerability.VulnerabilityIDs = append(vulnerability.Vulnerability.VulnerabilityIDs, vIDs.VulnerabilityID)
+			}
+
+			vulnerabilities = append(vulnerabilities, vulnerability)
+		}
+	}
+
+	return vulnerabilities, nil
+}
+
+func searchDependencies(ctx context.Context, gqlClient graphql.Client, pkgSpec model.PkgSpec, searchSoftware bool, startSBOM model.AllHasSBOMTree) (map[string]string, error) {
```

Comment:
Transitive dependency retrieval is already implemented in guacrest. Why not just reuse that?

Reply:
Yes, transitive dependency retrieval is already implemented in guacrest, but I wasn't sure whether that would get dependencies via the hasSBOMs IncludedSoftware.

Reply:
Yes it does. This is documented in the OpenAPI spec https://github.com/guacsec/guac/blob/9cfc2b7623a57d4116dc77cc6947af4d10c24705/pkg/guacrest/openapi.yaml#L34-L36, or see this test case https://github.com/guacsec/guac/blob/9cfc2b7623a57d4116dc77cc6947af4d10c24705/pkg/guacrest/server/retrieveDependencies_test.go#L55-L66

Reply:
Ok, I see what you mean. But, I have got a question about the functionality of RetrieveDependencies. How does RetrieveDependencies specify that we only want to search via includedSoftware (So that we are searching in the latest SBOM)?

Reply:
The dependency relationship isn't currently configurable beyond `linkCondition`. But more configuration options can certainly be added (see https://github.com/guacsec/guac/blob/9cfc2b7623a57d4116dc77cc6947af4d10c24705/pkg/guacrest/server/retrieveDependencies.go#L23). 

But I'm wondering what the intended output of the new dependencies endpoint is (see https://github.com/guacsec/guac/pull/2089#discussion_r1798610195)?

Reply:
Thank you, I see what you mean, and will be updating the code to have the latestsSBOM in the dependency search as well.

I have responded to the comment. I agree that having two different dependency searches and endpoints isn't needed, and the best way to go about this is to keep the two new endpoints, but use the retrieveDependencies.go code to actually search for dependencies.

Reply:
Just to clarify, the `retrieveDependencies` implementation does include the latest SBOM of each artifact or package. Are you referring to a dependency retrieval *exclusively* using the latest SBOM of each artifact and package?

Reply:
No, the `retrieveDependencies` implementation won't exclusively be using the latest SBOM, you can specify whether you want to use the latest sbom, but you can still do a regular search by purl or digest.

Reply:
The search on the latestSbom will be added in another PR.